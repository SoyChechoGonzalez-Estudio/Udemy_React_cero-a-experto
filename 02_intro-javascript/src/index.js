import {getHeroesById2} from './bases/08_import-export';

/*const promesa = new Promise((resolve, reject) => {
	setTimeout(() => {
		// Tarea
		// Importar
		const hero = getHeroesById2(3);
		// resolve(hero);
		resolve(hero);
		// reject('No se pudo encontrar el hÃ©roe');
	}, 2000);
});

promesa.then((hero) => {
	console.log(`Heroe ${hero}`);
})
	.catch((err) => {
		console.warn(err);
	});*/

const getHeroByIdAsync = () => {
	return new Promise((resolve, reject) => {
		setTimeout(() => {
			// const hero = getHeroesById2(3);
			// console.log(getHeroesById2(3));
			resolve(getHeroesById2(3));
		}, 2000);
	});
};

getHeroByIdAsync(2)
	.then((hero) => {
		console.log(`Heroe ${hero}`);
	});