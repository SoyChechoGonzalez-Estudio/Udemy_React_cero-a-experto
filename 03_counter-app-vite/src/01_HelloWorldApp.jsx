const HelloWorldApp = () => {
	return (
		<h1>Hola Mundo!!!!!!!!!!!</h1>
	);
};

export {HelloWorldApp};