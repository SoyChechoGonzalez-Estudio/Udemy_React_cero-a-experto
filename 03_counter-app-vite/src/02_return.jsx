const getResult = (a, b) => {
	return a + b;
};

const newMessage = {
	message: 'Hola, QuÃ© tal mi perro',
	title: 'Checho'
};
const FirstApp = () => {
	return (
		<>
			<h1>Hola Mundo</h1>
			<code>{JSON.stringify(newMessage)}</code>
			<h1>{getResult(1, 10)}</h1>
			<p>Soy un subtitulo</p>
		</>
	);
};

export {FirstApp};
