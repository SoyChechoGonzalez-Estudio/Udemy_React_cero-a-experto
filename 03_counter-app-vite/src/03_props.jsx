const FirstApp = ({title, subtitulo}) => {
	return (
		<>
			<h1>{title}</h1>
			<h1>{subtitulo + 1}</h1>
		</>
	);
};

export {FirstApp};