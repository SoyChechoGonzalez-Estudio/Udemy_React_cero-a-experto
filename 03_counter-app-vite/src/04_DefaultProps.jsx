import PropTypes from 'prop-types';

const FirstApp = ({title, subtitulo, name}) => {
	return (
		<>
			<h1>{title}</h1>
			<h1>{subtitulo + 1}</h1>
			<h1>{name}</h1>
		</>
	);
};

FirstApp.propTypes = {
	subtitulo: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
	name: 'Checho Gonz√°lez',
	subtitulo: 'No hay subtitulo',
	title: 'No hay titulo',
};
export {FirstApp};
