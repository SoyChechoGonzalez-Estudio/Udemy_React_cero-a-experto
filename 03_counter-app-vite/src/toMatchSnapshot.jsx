import PropTypes from 'prop-types';

const toMatchSnapshot = ({title, subtitle, name}) => {
	return (
		<>
			<h1>{title}</h1>
			<h1>{subtitle}</h1>
			<h1>{name}</h1>
		</>
	);
};

toMatchSnapshot.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string
};
toMatchSnapshot.defaultProps = {
	title: 'Checho Gonz√°lez',
	subtitle: 'No hay title',
	name: 'Checho'
};
export {toMatchSnapshot};