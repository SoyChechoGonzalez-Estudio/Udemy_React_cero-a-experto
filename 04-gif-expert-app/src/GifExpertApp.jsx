import {useState} from 'react';
import {AddCategory, GifGrid} from './Components';

const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One Punch']);
	const handleAddCategory = (newCategory) => {
		// Tarea
		// Agregar Algo
		if (categories.includes(newCategory)) return;
		// console.log(newCategory);
		setCategories([newCategory, ...categories]);
	};
	return (
		<>
			<h1>GifExpertApp</h1>
			
			<AddCategory
				onNewCategory={(event) => handleAddCategory(event)}
			/>
			
			{
				categories.map((category) => (
					<GifGrid
						key={category}
						category={category} />
				))
			}
		</>
	);
};

export {GifExpertApp};