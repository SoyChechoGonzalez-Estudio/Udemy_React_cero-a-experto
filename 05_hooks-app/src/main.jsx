import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
// import {HooksApp} from './HooksApp';
// import {CounterApp} from './01_useState/CounterApp';
// import {CounterWithCustomHook} from './01_useState/CounterWithCustomHook.jsx';
// import {SimpleForm} from './02_useEffect/SimpleForm';
// import {FormWithCustomHook} from './02_useEffect/FormWithCustomHook';
// import {MultipleCustomHooks} from './03_examples/MultipleCustomHooks';
// import {FocusScreen} from './04_useRef/FocusScreen';
// import {Layout} from './05_useLayoutEffect/Layout';
// import {Memorize} from './06_memos/Memorize';
// import {MemoHook} from './06_memos/MemoHook';
// import {CallbackHook} from './06_memos/CallbackHook';
import {Padre} from './07-tarea-memo/Padre';

ReactDOM.createRoot(document.getElementById('root')).render(
	// <React.StrictMode>
	<Padre />
	// </React.StrictMode>
);
