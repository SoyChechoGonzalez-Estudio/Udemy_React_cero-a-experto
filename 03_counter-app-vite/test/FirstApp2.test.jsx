import {render, screen} from '@testing-library/react';
import {FirstApp} from '../src/FirstApp';

describe('Pruebas en <FirstApp/>', () => {
	const title = 'Hola, soy Gokú';
	const subtitle = 'Soy un subtitulo';
	test('Debe hacer match con el snapshot', () => {
		const {container} = render(
			<FirstApp title={title} />
		);
		expect(container).toMatchSnapshot();
	});
	test('Debe mostrart el mensaje "Hola, soy Gokú"', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByText(title)).toBeTruthy();
		// screen.debug();
	});
	test('Debe de mostrar el título en un H1', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
	});
	test('Debe de mostrar el subtítulo enviado por props', () => {
		render(
			<FirstApp title={title}
			          subtitle={subtitle}
			/>
		);
		expect(screen.getAllByText(subtitle).length).toBe(2);
		
	});
	
});