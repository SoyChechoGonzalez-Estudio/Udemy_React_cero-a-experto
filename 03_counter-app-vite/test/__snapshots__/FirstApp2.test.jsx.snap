// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Gok√∫
  </h1>
  <h1>
    No hay subtitle
  </h1>
  <h1>
    No hay subtitle
  </h1>
  <h1>
    Checho
  </h1>
</div>
`;
